src/commands/cluster.ts

```diff
@@ -70,6 +70,15 @@ export class DeleteClusterCommandHandler {
         if (!clusterId) {
             return;
         }
+        const cluster = this.clusterSettings.get(clusterId);
+        if (!cluster) {
+            return;
+        }
+
+        const deleteConfirmation: string | undefined = await vscode.window.showWarningMessage(`Are you sure you want to delete cluster '${cluster.name}'?`, 'Cancel', 'Delete');
```

Comment:
You can loose the explicit typings (string | undefined) - not used in other places